### PR TITLE
[orchestrator] Creates token for uploading user artifacts.

### PR DESCRIPTION
**Pull request by ser-io**

### CODE REVIEW

File: frontend/src/host_orchestrator/orchestrator/userartifacts.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		UserArtifactsManagerOpts: opts,
+	}
+}
+
+func (m *UserArtifactsManagerImpl) NewToken() (*apiv1.UploadToken, error) {
+	if err := createDir(m.Dir, false); err != nil {
+		return nil, err
+	}
+	token := m.UUIDFactory()
+	if err := createDir(m.Dir+"/"+token, false); err != nil {
```
Comment (jemoreira) [inline frontend/src/host_orchestrator/orchestrator/userartifacts.go:51]: Shouldn't this fail if the directory already exists?

Also, take a look at [Call site readability and boolean parameters](https://g3doc.corp.google.com/devtools/library_club/g3doc/totw/94.md?cl=head)

File: frontend/src/host_orchestrator/orchestrator/userartifacts.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+// Creates a new instance of UserArtifactsManagerImpl.
+func NewUserArtifactsManagerImpl(opts UserArtifactsManagerOpts) UserArtifactsManager {
+	return &UserArtifactsManagerImpl{
+		UserArtifactsManagerOpts: opts,
+	}
+}
+
+func (m *UserArtifactsManagerImpl) NewToken() (*apiv1.UploadToken, error) {
```
[the remark this answers is not in the document]
Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/userartifacts.go]: true, I also removed the term `token` from this context and it's confusing and started using the term `directory` instead.

changes to NameFactory as well

File: frontend/src/host_orchestrator/orchestrator/userartifacts.go (hunk 1 above)
Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/userartifacts.go]: good catch! done.

[Call site readability and boolean parameters](https://g3doc.corp.google.com/devtools/library_club/g3doc/totw/94.md?cl=head) was a great reading, will make the relevant changes afterwards.

**Review by jemoreira — APPROVED** [review]

File: frontend/src/host_orchestrator/main.go [hunk 3]
```diff
@@ -137,6 +139,11 @@ func main() {
 		HostValidator:    &orchestrator.HostValidator{ExecContext: exec.Command},
 	}
 	im := orchestrator.NewCVDToolInstanceManager(&opts)
+	uamOpts := orchestrator.UserArtifactsManagerOpts{
+		RootDir:      imRootDir + "/user_artifacs",
+		NamesFactory: func() string { return uuid.New().String() },
```
[the remark this answers is not in the document]
Comment (ser-io) [reply frontend/src/host_orchestrator/main.go]: done

**Pull request merged**